Monitor/Monitor.h

```diff
@@ -774,7 +774,7 @@ namespace Plugin {
                             // Moreover it's the only only which now becomes active. This means probing
                             // has to be activated as well since it was stopped at point the last observee
                             // turned inactive
-                            _job.Submit();
+                            _job.Schedule(Core::Time::Now());
```

Comment:
I understand the deadlock situation. But this solution will no longer work in R3 as there if the schedule_time <= Now() it will internally submit instead of schedule.
I think an alternative and more consistent way to fix it is to Submit the job outside the lock as we do elsewhere in this file. I think that is the bug, that the submit is inside the lock, that should not have happened

Reply:
@MFransen69, the lock with is causing the deadlock is `_notificationLock.Lock();` in ServiceMap::StateChange.

Reply:
Hi @npoltorapavlo 
Thanks for explaining. The way the state changes are handled is redesigned in R3. Let me discuss with the person who did this and get back on this (he is not in at the moment)